File: src/main/java/net/ndrei/teslapoweredthingies/gui/ItemStackPiece.java
package net.ndrei.teslapoweredthingies.gui;

import net.minecraft.client.renderer.GlStateManager;
import net.minecraft.client.renderer.RenderHelper;
import net.minecraft.item.ItemStack;
import net.ndrei.teslacorelib.compatibility.FontRendererUtil;
import net.ndrei.teslacorelib.compatibility.ItemStackUtil;
import net.ndrei.teslacorelib.gui.BasicContainerGuiPiece;
import net.ndrei.teslacorelib.gui.BasicTeslaGuiContainer;

/**
 * Created by CF on 2017-01-08.
 */
public class ItemStackPiece extends BasicContainerGuiPiece {
    private IWorkItemProvider provider;

    public ItemStackPiece(int left, int top, int width, int height, IWorkItemProvider provider) {
        super(left, top, width, height);

        this.provider = provider;
    }

    @Override
    public void drawMiddleLayer(BasicTeslaGuiContainer container, int guiX, int guiY, float partialTicks, int mouseX, int mouseY) {
        ItemStack stack = (this.provider == null) ? ItemStack.EMPTY : this.provider.getWorkItem();
        if (!ItemStackUtil.isEmpty(stack)) {
            int x = this.getLeft() + (this.getWidth() - 16) / 2 + 1;
            int y = this.getTop() + (this.getHeight() - 16) / 2;

            RenderHelper.enableGUIStandardItemLighting();
            GlStateManager.pushMatrix();
            GlStateManager.enableDepth();
            container.getItemRenderer().renderItemAndEffectIntoGUI(stack, guiX + x, guiY + y);
            container.getItemRenderer().renderItemOverlayIntoGUI(FontRendererUtil.getFontRenderer(), stack, x, y, null);
            GlStateManager.popMatrix();
            RenderHelper.disableStandardItemLighting();
        }
    }
}

File: src/main/java/net/ndrei/teslapoweredthingies/gui/ItemStackPiece.kt
package net.ndrei.teslapoweredthingies.gui

import net.minecraft.client.renderer.GlStateManager
import net.minecraft.client.renderer.RenderHelper
import net.minecraft.item.ItemStack
import net.ndrei.teslacorelib.compatibility.FontRendererUtil
import net.ndrei.teslacorelib.compatibility.ItemStackUtil
import net.ndrei.teslacorelib.gui.BasicContainerGuiPiece
import net.ndrei.teslacorelib.gui.BasicTeslaGuiContainer

/**
 * Created by CF on 2017-06-30.
 */
open class ItemStackPiece(left: Int, top: Int, width: Int, height: Int, private val provider: IWorkItemProvider?)
    : BasicContainerGuiPiece(left, top, width, height) {

    override fun drawMiddleLayer(container: BasicTeslaGuiContainer<*>, guiX: Int, guiY: Int, partialTicks: Float, mouseX: Int, mouseY: Int) {
        val stack = if (this.provider == null) ItemStack.EMPTY else this.provider.workItem
        if (!ItemStackUtil.isEmpty(stack)) {
            val x = this.left + (this.width - 16) / 2
            val y = this.top + (this.height - 16) / 2

            RenderHelper.enableGUIStandardItemLighting()
            GlStateManager.pushMatrix()
            GlStateManager.enableDepth()
            container.itemRenderer.renderItemAndEffectIntoGUI(stack, guiX + x, guiY + y)
            container.itemRenderer.renderItemOverlayIntoGUI(FontRendererUtil.fontRenderer, stack, x, y, null)
            GlStateManager.popMatrix()
            RenderHelper.disableStandardItemLighting()
        }
    }
}
